packages/components/src/types/common.ts

```diff
@@ -0,0 +1,13 @@
+export type TextWithA11y = {
+  /** Text to display */
+  text: string
+  /** Accessibility override for text */
+  a11y?: string
+}
+
+export type TextWithA11yAndValue = TextWithA11y & {
+  /** Value that may differ from provided text */
+  value?: string | number
+}
+
+export type StringOrTextWithA11y = string | TextWithA11y
```

Comment:
I'm not sure this adds value as a named type. 

On the one hand, it provides a "do not repeat yourself" value. On the other hand, it's so specific that it couldn't be flexible in the future without renaming anyway so I'm not sure it adds meaningful value vs just having `string | TextWithA11y` typed on the props themselves.

Reply:
Caveat: it may make more sense if we were exporting it for consumer consumption so when they were nesting things and passing along the typing they could use one thing instead of mimicking our `string | TextWithA11y`.

Reply:
Seeing how many places it's used working through the rest of the review--perhaps worth keeping even if not exported.